### DISCUSSION TITLE
[Share] Write Zotero Item Key into Extra Field

### DISCUSSION THREAD

**Discussion by siocean** [windingwind/zotero-actions-tags#393] [Action Scripts]
### Description

Writes the Zotero Item Key into the Extra Field of the item. 
- To batch apply, select all items and select the trigger.
- Set Event to create Item so that newly created items key is written in it's Extra field.
- Removes the existing content in the Extra field.

Use cases:
- The item key value can be included in the attachment file name by renaming as {{extra}}

### Event

Creat Item

### Operation

Script

### Data

```javascript
/**
 * Write Zotero Item Key into Extra Field.
 * @author siocean
 * @usage Applies to selected items upon trigger or newly created items if the event Create Item is selected
 * @link https://github.com/windingwind/zotero-actions-tags/discussions/
 * @see https://github.com/windingwind/zotero-actions-tags/discussions/
 */

var ZoteroPane = Zotero.getActiveZoteroPane();
var items = ZoteroPane.getSelectedItems();

if (!items.length) {
    return "No items selected";
}

await Zotero.DB.executeTransaction(async function () {
    for (let item of items) {
        if (item.isRegularItem()) {
            let itemKey = item.key;
            item.setField('extra', itemKey);
            await item.save();
        }
    }
});
return items.length + " item(s) updated";
```

### Anything else

Suggestions and improvement are most welcome and appreciated!

**Comment by windingwind** [comment]
https://github.com/windingwind/zotero-actions-tags?tab=readme-ov-file#custom-script

Please always use the items instead of ZoteroPane.getSelectedItems(). The action can be triggered from entrances outside the library, where the items are not from ZoteroPane.

**Comment by windingwind** [comment]
Also, please don’t use the extra field in this way, see https://www.zotero.org/support/kb/item_types_and_fields#citing_fields_from_extra
